src/prune_builds

```diff
@@ -58,7 +62,10 @@ if args.insert_only:
 
 skip_pruning = (args.keep_last_n == 0)
 
-# first, regen builds.json
+# collect all builds being pointed to by tags
+tagged_builds = set([tag['target'] for tag in builddata.get('tags', [])])
```

Comment:
Would it be safer to do a `tag.get('target', "")` or is it guaranteed to be present?

Reply:
Hmm, I think I'd rather we error out if it's somehow not present since we only ever create tags with targets. If later, we somehow have a use case for that, that's likely something that'd need to be handled elsewhere in this script and in `src/cmd-tag`.

Reply:
Fair enough. If we ever encounter the case and want to handle it in the future we can always add a more verbose error.